docker-compose/setup-script/noco.sh

```diff
@@ -477,37 +532,143 @@ server {
 EOF
 fi
 
-IS_DOCKER_COMPOSE_REQUIRE_SUDO=$(check_for_docker_compose_sudo)
+IS_DOCKER_REQUIRE_SUDO=$(check_for_docker_sudo)
+DOCKER_COMMAND=$([ "$IS_DOCKER_REQUIRE_SUDO" = "y" ] && echo "sudo docker" || echo "docker")
 
+# Generate help script
+cat > ./help.sh <<EOF
+#!/bin/bash
 
-# Generate the update.sh file for upgrading images
-if [ "$IS_DOCKER_COMPOSE_REQUIRE_SUDO" = "y" ]; then
-  cat > ./update.sh <<EOF
-sudo docker-compose pull
-sudo docker-compose up -d --force-recreate
-sudo docker image prune -a -f
-EOF
-else
-  cat > ./update.sh <<EOF
-docker-compose pull
-docker-compose up -d --force-recreate
-docker image prune -a -f
+trap show_menu INT
+
+
+$(declare -f read_number)
+
+$(declare -f read_number_range)
+
+# Function to display the menu
+show_menu() {
+    clear
+    echo ""
+    echo \$MSG
+    echo "Service Management Menu:"
+    echo "1. Start Service"
+    echo "2. Stop Service"
+    echo "3. Logs"
+    echo "4. Restart"
+    echo "5. Upgrade"
+    echo "6. Scale"
+    echo "7. Monitoring"
+    echo "0. Exit"
+}
+
+# Function to start the service
+start_service() {
+    echo -e "\nStarting nocodb..."
+    $DOCKER_COMMAND compose up -d
+}
+
+# Function to stop the service
+stop_service() {
+    echo -e "\nStopping nocodb..."
+    $DOCKER_COMMAND compose stop
+}
+
+# Function to show logs
+show_logs() {
+    clear
+    echo "Select a container for logs:"
+    echo "1. nocodb"
+    echo "2. db"
+    echo "3. nginx"
+    echo "4. redis"
+    echo "5. watchtower"
+    echo "6. All"
+    echo "0. Back to main menu"
+    echo "Enter your choice: "
+
+    read -n 1 log_choise
+
+    case \$log_choise in
+        1) $DOCKER_COMMAND compose logs -f nocodb ;;
+        2) $DOCKER_COMMAND compose logs -f db ;;
+        3) $DOCKER_COMMAND compose logs -f nginx ;;
+        4) $DOCKER_COMMAND compose logs -f redis ;;
+        5) $DOCKER_COMMAND compose logs -f watchtower ;;
+        6) $DOCKER_COMMAND compose logs -f ;;
+        0) return ;;
+        *) echo "Invalid choice. Returning to main menu." ;;
+    esac
```

Comment:
If possible we can make these part dynamic based on users choice selection since few services are optional.

Reply:
I mentioned to @rohittp0 that if we can somehow show docker stats containers with their name and user can select that to show the logs.

Reply:
May be we can use `docker-compose config` - https://github.com/docker/compose/issues/2683

Reply:
Now it reads the container info and also accounts for replicas,

```markdown
Select a container for logs:
1. db (1 replicas)
2. nginx (1 replicas)
3. nocodb (3 replicas)
4. redis (1 replicas)
5. watchtower (1 replicas)
A. All
0. Back to main menu
Enter your choice: 
```

```markdown
Select a replica for nocodb:
1. nocodb replica 1
2. nocodb replica 2
3. nocodb replica 3
A. All
Enter replica number: 
```